api/src/main/java/ai/djl/training/optimizer/Sgd.java

```diff
@@ -56,6 +57,9 @@ protected Sgd(Builder builder) {
     /** {@inheritDoc} */
     @Override
     public void update(String parameterId, NDArray weight, NDArray grad) {
+        if (learningRateTracker instanceof FixedPerVarTracker) {
+            ((FixedPerVarTracker) learningRateTracker).setParameterId(parameterId);
```

Comment:
It isn't great to use a system like this to pass a special case argument of parameterId here. Instead, maybe we could add a new kind of tracker named something like `ParameterTracker`. The ParamterTracker can require the parameter to get a new value, and all standard Trackers would also be a `ParameterTracker` if we have `Tracker extend ParameterTracker`. If we leave it as a special case like this, it would make it difficult if a new tracker that also required parameters were created or if new optimizers were created without the special handling for the FixedPerVarTracker